tests/integration/components/filter-menu-test.js

```diff
@@ -0,0 +1,18 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { findAll, render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | filter-menu', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test(`it renders and it's not empty`, async function(assert) {
+    await render(hbs`<FilterMenu />`);
+    assert.notEqual(this.element.textContent.trim(), '');
+  });
+
+  test('buttons were rendered', async function(assert) {
+    await render(hbs`<FilterMenu />`);
+    assert.ok(findAll('[data-test-link]').length > 0);
+  });
```

Comment:
When I check for the presence of multiple items, I like to first use `findAll` to get these items, then check for the count and test either some or all items.

```suggestion
  test('renders 11 links', async function(assert) {
    await render(hbs`
      <FilterMenu />
    `);

    const links = findAll('[data-test-link]');

    assert.strictEqual(
      links.length,
      11,
      'We see 11 links.'
    );

    assert.dom(links[0])
      .hasText(
        'adopted-ember-addons',
        'We see the correct 1st link.'
      );
  });

    assert.dom(links[links.length - 1])
      .hasText(
        'RFCs',
        'We see the correct last link.'
      );
  });
```

Again, the assertions above are stronger than the assertion `assert.ok(findAll('[data-test-link]').length > 0);` The latter assertion can pass if the component renders an incorrect number of links, whereas the suggested assertions will fail.